lib/house.rb

```diff
@@ -0,0 +1,49 @@
+class House
+  attr_reader :price, :address, :rooms, :details
+
+  def initialize(price, address)
+    @price = price[1..-1].to_i
+    @address = address
+    @rooms = []
+    @total_area = 0
+    @details = {"price" => @price,
+                "address" => @address
+               }
+  end
+
+  def add_room(room_object)
+    @rooms << room_object
+  end
+
+  def above_market_average?
+    if @price > 500000 then true else false end
+  end
+
+  def rooms_from_category(category)
+    rooms_in_category = []
+    @rooms.each do |room|
+      if room.category == category
+        rooms_in_category << room
+      end
+    end
+    rooms_in_category
+  end
+
+  def area
+    @rooms.sum {|room| room.area}
+  end
+
+  def price_per_square_foot
+    (@price / area.to_f).round(2)
+  end
+
+  def rooms_sorted_by_area
+    @rooms.sort_by(&:area).reverse!
```

Comment:
elegant solution here.  In general, we try to avoid permanently altering the original collection unless we have a very good reason, which means we steer clear of the `!` operator. This would work just fine with the non-destructive form of `reverse`.